src/util/std_expr.h

```diff
@@ -2962,6 +2962,17 @@ class member_exprt:public exprt
   {
     return op0();
   }
+
+  //Retrieves the object(symbol) this member corresponds to
+  inline const symbol_exprt &symbol() const
+  {
+    const exprt &op=op0();
+    if (op.id()==ID_member)
+    {
+      return static_cast<const member_exprt &>(op).symbol();
+    }
+    return to_symbol_expr(op);
+  }
```

Comment:
The root object need not be a symbol - it could, for example, be an array. Use object_descriptor_exprt to compute the root_object().